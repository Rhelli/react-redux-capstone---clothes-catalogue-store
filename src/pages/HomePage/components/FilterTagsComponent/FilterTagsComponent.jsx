/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React from 'react';
import PropTypes from 'prop-types';
import styles from './FilterTagsComponent.module.scss';

const FilterTagsComponent = props => {
  const { enableSelectedFilter } = props;

  return (
    <div className={styles.filterTagsContainer}>
      <div className={styles.filterCategory}>
        <h4 className={styles.categoryTitles}>Gender</h4>
        <div className={styles.categoryTags}>
          <div
            className={styles.tag}
            data-name="male"
            onClick={event => enableSelectedFilter(event, 'genderFilter')}
          >
            <h6 data-name="male">Men&apos;s</h6>
          </div>
          <div
            className={styles.tag}
            data-name="female"
            onClick={event => enableSelectedFilter(event, 'genderFilter')}
          >
            <h6 data-name="female">Women&apos;s</h6>
          </div>
        </div>
      </div>
      <div className={styles.filterCategory}>
        <h4 className={styles.categoryTitles}>Clothes</h4>
        <div className={styles.categoryTags}>
          <div
            data-name="jacket"
            onClick={event => enableSelectedFilter(event, 'clothesFilter')}
          >
            <h6 data-name="jacket">Coats & Jackets</h6>
          </div>
          <div
            data-name="jeans"
            onClick={event => enableSelectedFilter(event, 'clothesFilter')}
          >
            <h6 data-name="jeans">Jeans</h6>
          </div>
          <div
            data-name="shirts"
            onClick={event => enableSelectedFilter(event, 'clothesFilter')}
          >
            <h6 data-name="shirts">Shirts</h6>
          </div>
          <div
            data-name="tops"
            onClick={event => enableSelectedFilter(event, 'clothesFilter')}
          >
            <h6 data-name="tops">Tops</h6>
          </div>
          <div
            data-name="dresses"
            onClick={event => enableSelectedFilter(event, 'clothesFilter')}
          >
            <h6 data-name="dresses">Dresses</h6>
          </div>
        </div>
      </div>
      <div className={styles.filterCategory}>
        <h4 className={styles.categoryTitles}>Color</h4>
        <div className={styles.categoryTags}>
          <div
            data-name="black"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="black">Black</h6>
          </div>
          <div
            data-name="darkestSpruce"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="darkestSpruce">Darkest Spruce</h6>
          </div>
          <div
            data-name="multi"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="multi">Multi</h6>
          </div>
          <div
            data-name="blue"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="blue">Blue</h6>
          </div>
          <div
            data-name="blueDenim"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="blueDenim">Blue Denim</h6>
          </div>
          <div
            data-name="navy"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="navy">Navy</h6>
          </div>
          <div
            data-name="green"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="green">Green</h6>
          </div>
          <div
            data-name="white"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="white">White</h6>
          </div>
          <div
            data-name="grape"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="grape">Grape</h6>
          </div>
          <div
            data-name="khaki"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="khaki">Khaki</h6>
          </div>
          <div
            data-name="grey"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="grey">Grey</h6>
          </div>
          <div
            data-name="vintageIndigo"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="vintageIndigo">Vintage Indigo</h6>
          </div>
          <div
            data-name="mineralBlue"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="mineralBlue">Mineral Blue</h6>
          </div>
          <div
            data-name="charcoal"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="charcoal">Charcoal</h6>
          </div>
          <div
            data-name="tan"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="tan">Tan</h6>
          </div>
          <div
            data-name="dustyBlue"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="dustyBlue">Dusty Blue</h6>
          </div>
          <div
            data-name="midwash"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="midwash">Midwash</h6>
          </div>
          <div
            data-name="red"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="red">Red</h6>
          </div>
          <div
            data-name="rose"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="rose">Rose</h6>
          </div>
          <div
            data-name="lilac"
            onClick={event => enableSelectedFilter(event, 'colorFilter')}
          >
            <h6 data-name="lilac">Lilac</h6>
          </div>
        </div>
      </div>
    </div>
  );
};

FilterTagsComponent.propTypes = {
  enableSelectedFilter: PropTypes.func.isRequired,
};

export default FilterTagsComponent;
