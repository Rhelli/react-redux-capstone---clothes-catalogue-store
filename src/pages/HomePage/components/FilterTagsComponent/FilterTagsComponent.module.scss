.filterTagsContainer {
  display: flex;
  flex-flow: column nowrap;
  width: 100%;
  height: 40vh;
  margin-bottom: 100px;
}

.filterCategory {
  display: flex;
  flex-flow: column nowrap;
  width: 100%;
  border: 1px solid red;
}

.categoryTags {
  display: flex;
  flex-flow: row nowrap;
  width: auto;
}

.colorCategory {
  display: flex;
  flex-flow: column wrap;
  width: 50%;
}

.tag {
  width: auto;
}
