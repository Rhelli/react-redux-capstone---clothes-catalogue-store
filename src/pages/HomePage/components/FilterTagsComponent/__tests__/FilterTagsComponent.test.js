import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import FilterTagsComponent from '../FilterTagsComponent';

const enableSelectedFilter = jest.fn();
const filterTagsComponent = <FilterTagsComponent enableSelectedFilter={enableSelectedFilter} />;

describe('The Appearance and Functionality of the FilterTagsComponent', () => {
  it('Matches the FilterTagsComponent Snapshot', () => {
    const tree = renderer.create(filterTagsComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Contains three main containers for each filter category', () => {
    const { container } = render(filterTagsComponent);
    expect(container.firstChild.childElementCount).toEqual(3);
  });

  it('Contains the correct number of filter tags available in each category', () => {
    const { container } = render(filterTagsComponent);
    expect(container.firstChild.firstChild.lastChild.childElementCount).toEqual(2);
    expect(container.firstChild.firstChild.nextSibling.lastChild.childElementCount).toEqual(5);
    expect(container.firstChild.lastChild.lastChild.childElementCount).toEqual(20);
  });
});
