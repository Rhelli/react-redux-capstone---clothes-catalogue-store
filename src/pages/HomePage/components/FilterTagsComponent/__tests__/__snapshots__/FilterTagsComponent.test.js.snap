// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Appearance and Functionality of the FilterTagsComponent Matches the FilterTagsComponent Snapshot 1`] = `
<div
  className="filterTagsContainer"
>
  <div
    className="filterCategory"
  >
    <h4
      className="categoryTitles"
    >
      Gender
    </h4>
    <div
      className="categoryTags"
    >
      <div
        className="male"
        data-name="male"
        onClick={[Function]}
      >
        <h6
          data-name="male"
        >
          Men's
        </h6>
      </div>
      <div
        className="female"
        data-name="female"
        onClick={[Function]}
      >
        <h6
          data-name="female"
        >
          Women's
        </h6>
      </div>
    </div>
  </div>
  <div
    className="filterCategory"
  >
    <h4
      className="categoryTitles"
    >
      Clothes
    </h4>
    <div
      className="categoryTags"
    >
      <div
        className="jacket"
        data-name="jacket"
        onClick={[Function]}
      >
        <h6
          data-name="jacket"
        >
          Coats & Jackets
        </h6>
      </div>
      <div
        className="jeans"
        data-name="jeans"
        onClick={[Function]}
      >
        <h6
          data-name="jeans"
        >
          Jeans
        </h6>
      </div>
      <div
        className="shirts"
        data-name="shirts"
        onClick={[Function]}
      >
        <h6
          data-name="shirts"
        >
          Shirts
        </h6>
      </div>
      <div
        className="tops"
        data-name="tops"
        onClick={[Function]}
      >
        <h6
          data-name="tops"
        >
          Tops
        </h6>
      </div>
      <div
        className="dresses"
        data-name="dresses"
        onClick={[Function]}
      >
        <h6
          data-name="dresses"
        >
          Dresses
        </h6>
      </div>
    </div>
  </div>
  <div
    className="filterCategory"
  >
    <h4
      className="categoryTitles"
    >
      Colour
    </h4>
    <div
      className="categoryTags"
    >
      <div
        data-name="black"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#101010",
            "borderBottom": "2px solid #202020",
            "color": "#C3C3C3",
          }
        }
      >
        <h6
          data-name="black"
        >
          Black
        </h6>
      </div>
      <div
        data-name="darkestSpruce"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#0E6251",
            "borderBottom": "2px solid #0C5647",
            "color": "#A5C4BE",
          }
        }
      >
        <h6
          data-name="darkestSpruce"
        >
          Darkest Spruce
        </h6>
      </div>
      <div
        className="multi"
        data-name="multi"
        onClick={[Function]}
      >
        <h6
          data-name="multi"
        >
          Multi
        </h6>
      </div>
      <div
        data-name="blue"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#3498DB",
            "borderBottom": "2px solid #2E85C0",
            "color": "#CCE5F6",
          }
        }
      >
        <h6
          data-name="blue"
        >
          Blue
        </h6>
      </div>
      <div
        data-name="blueDenim"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#546E7A",
            "borderBottom": "2px solid #4A606B",
            "color": "#D4DBDE",
          }
        }
      >
        <h6
          data-name="blueDenim"
        >
          Blue Denim
        </h6>
      </div>
      <div
        data-name="navy"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#2C3E50",
            "borderBottom": "2px solid #212F3C",
            "color": "#CACFD3",
          }
        }
      >
        <h6
          data-name="navy"
        >
          Navy
        </h6>
      </div>
      <div
        data-name="green"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#229954",
            "borderBottom": "2px solid #1A733F",
            "color": "#C8E6D4",
          }
        }
      >
        <h6
          data-name="green"
        >
          Green
        </h6>
      </div>
      <div
        data-name="white"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderBottom": "2px solid #DFDFDF",
            "color": "#303030",
          }
        }
      >
        <h6
          data-name="white"
        >
          White
        </h6>
      </div>
      <div
        data-name="grape"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#884EA0",
            "borderBottom": "2px solid #663B78",
            "color": "#E1D3E7",
          }
        }
      >
        <h6
          data-name="grape"
        >
          Grape
        </h6>
      </div>
      <div
        data-name="khaki"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#AED581",
            "borderBottom": "2px solid #83A061",
            "color": "#6D8551",
          }
        }
      >
        <h6
          data-name="khaki"
        >
          Khaki
        </h6>
      </div>
      <div
        data-name="grey"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#BFC9CA",
            "borderBottom": "2px solid #8F9798",
            "color": "#777E7E",
          }
        }
      >
        <h6
          data-name="grey"
        >
          Grey
        </h6>
      </div>
      <div
        data-name="vintageIndigo"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#7986CB",
            "borderBottom": "2px solid #5B6598",
            "color": "#DDE1F2",
          }
        }
      >
        <h6
          data-name="vintageIndigo"
        >
          Vintage Indigo
        </h6>
      </div>
      <div
        data-name="mineralBlue"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#80DEEA",
            "borderBottom": "2px solid #60A6AF",
            "color": "#406F75",
          }
        }
      >
        <h6
          data-name="mineralBlue"
        >
          Mineral Blue
        </h6>
      </div>
      <div
        data-name="charcoal"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#424242",
            "borderBottom": "2px solid #292929",
            "color": "#D0D0D0",
          }
        }
      >
        <h6
          data-name="charcoal"
        >
          Charcoal
        </h6>
      </div>
      <div
        data-name="tan"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#CA6F1E",
            "borderBottom": "2px solid #985316",
            "color": "#F2DBC7",
          }
        }
      >
        <h6
          data-name="tan"
        >
          Tan
        </h6>
      </div>
      <div
        data-name="dustyBlue"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#AED6F1",
            "borderBottom": "2px solid #83A1B5",
            "color": "#6D8697",
          }
        }
      >
        <h6
          data-name="dustyBlue"
        >
          Dusty Blue
        </h6>
      </div>
      <div
        data-name="midwash"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#7FB3D5",
            "borderBottom": "2px solid #5F86A0",
            "color": "#DFECF5",
          }
        }
      >
        <h6
          data-name="midwash"
        >
          Midwash
        </h6>
      </div>
      <div
        data-name="red"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#E53935",
            "borderBottom": "2px solid #AC2B28",
            "color": "#F8CECC",
          }
        }
      >
        <h6
          data-name="red"
        >
          Red
        </h6>
      </div>
      <div
        data-name="rose"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#EB9696",
            "borderBottom": "2px solid #B07171",
            "color": "#FAE5E5",
          }
        }
      >
        <h6
          data-name="rose"
        >
          Rose
        </h6>
      </div>
      <div
        data-name="lilac"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#E1BEE7",
            "borderBottom": "2px solid #A98FAD",
            "color": "#F7EFF9",
          }
        }
      >
        <h6
          data-name="lilac"
        >
          Lilac
        </h6>
      </div>
    </div>
  </div>
</div>
`;
